### PR TITLE
SNP: add support to read UVM security context from files rather than environment

### LINKED ISSUE #5164: Update ACI SEV-SNP configuration from files
Following https://github.com/microsoft/hcsshim/pull/1708, we'll need to update https://microsoft.github.io/CCF/main/operations/configuration.html#environment as the UVM reference info, VCEK certificates and security policy will be soon specified via files rather than environment variables.

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves #5164 

CCF nodes can now read the UVM security context from files as well as from environment variables, as per the recent changes made to confidential ACI.

### CODE REVIEW

File: .azure-pipelines-templates/deploy_aci.yml [hunk 1]
```diff
@@ -182,63 +182,63 @@ jobs:
...
-          if [[ ${{ parameters.secondaries.count }} != 0 ]]; then
-            python3.8 scripts/azure_deployment/arm_template.py remove aci \
-            --subscription-id $(CCF_AZURE_SUBSCRIPTION_ID) \
-            --resource-group ccf-aci \
-            --aci-type dynamic-agent \
-            --deployment-name ci-$(Build.BuildNumber)-secondaries
-          fi
-        name: cleanup_secondary_acis
-        displayName: "Delete the secondary ACIs and Azure Deployments"
+  # - job: cleanup_aci
```
Comment (achamayou) [inline .azure-pipelines-templates/deploy_aci.yml:185]: Presumably needs to be commented back in before merging?

File: scripts/azure_deployment/arm_aci.py [hunk 2]
```diff
@@ -34,15 +34,16 @@ def get_pubkey():
 
 
 def setup_environment_command():
-    # ACI SEV-SNP environment variables are only set for PID 1 (i.e. container's command)
+    # ACI SEV-SNP environment variables are only set for PID the container's command
```
Comment (achamayou) [inline scripts/azure_deployment/arm_aci.py:37]:
```suggestion
    # ACI SEV-SNP environment variables are only set for the PID of the container's command
```

**Review by achamayou — APPROVED** [review]

**Comment by jumaffre** [comment]
Marking this PR as draft as I want to make sure it still works in North Europe with the old way of setting the security context via envvars.

File: doc/host_config_schema/cchost_config.json [hunk 3]
```diff
@@ -448,6 +448,10 @@
         "environment": {
           "type": "object",
           "properties": {
+            "security_context_directory": {
+              "type": ["string", "null"],
+              "description": "Name of environment variable (e.g. ``UVM_SECURITY_CONTEXT_DIR``) specifying the directory containing the security context files (i.e. ``host-amd-cert-base64``, ``security-policy-base64`` and ``reference-info-base64``). If this is set, ``report_endorsements``, ``security_policy`` and ``uvm_endorsements`` are ignored"
```
[the remark this answers is not in the document]
Comment (jumaffre) [reply doc/host_config_schema/cchost_config.json:448]: Unfortunately, I cannot find any mention of these variables in any public documentation. This hopefully gets addressed before GA. In the meantime, [this PR](https://github.com/microsoft/hcsshim/pull/1708) (linked in #5164) should be used as reference.

**Review by achamayou — APPROVED** [review]

**Pull request merged**